packages/entity/src/EntityAssociationLoader.ts

```diff
@@ -54,14 +55,20 @@ export default class EntityAssociationLoader<
       .getViewerScopedEntityCompanionForClass(associatedEntityClass)
       .getQueryContextProvider()
       .getQueryContext()
-  ): Promise<Result<TAssociatedEntity>> {
+  ): Promise<
+    Result<null extends TFields[TIdentifyingField] ? TAssociatedEntity | null : TAssociatedEntity>
+  > {
     const associatedEntityID = this.entity.getField(fieldIdentifyingAssociatedEntity);
+    if (!associatedEntityID) {
+      return result(null) as any;
```

Comment:
Shouldn't a sufficiently knowledgeable type checker know that if `TFields[TIdentifyingField]` is nullable, then `this.entity.getField(fieldIdentifyingAssociatedEntity)` is nullable and thus returning `result(null)` is OK? If so, we might want to use `@ts-expect-error` instead of `any` here.

Is the issue that `Result<null>` doesn't satisfy `Result<TAssociatedEntity | null>`? Maybe we need some `readonly`s in Result's definition for covariance?

Reply:
It's hard to tell how the typechecker is analyzing this, but the full message is:
```
src/EntityAssociationLoader.ts(63,7): error TS2322: Type 'Success<null>' is not assignable to type 'Result<null extends TFields[TIdentifyingField] ? TAssociatedEntity | null : TAssociatedEntity>'.
  Type 'Success<null>' is not assignable to type 'Success<null extends TFields[TIdentifyingField] ? TAssociatedEntity | null : TAssociatedEntity>'.
    Type 'null' is not assignable to type 'null extends TFields[TIdentifyingField] ? TAssociatedEntity | null : TAssociatedEntity'.
```
so it looks like it's unable to use the knowledge about `this.entity.getField(fieldIdentifyingAssociatedEntity)` when analyzing `return result(null);`.

Reply:
I'll go ahead and use `@ts-expect-error`.